src/goto-instrument/contracts/dynamic-frames/dfcc_cfg_info.cpp

```diff
@@ -0,0 +1,894 @@
+/*******************************************************************\
+
+Module: Dynamic frame condition checking for function and loop contracts.
+
+Author: Qinheping Hu, qinhh@amazon.com
+Author: Remi Delmas delmasrd@amazon.com
+
+Date: March 2023
+
+\*******************************************************************/
+
+#include "dfcc_cfg_info.h"
+
+#include <util/c_types.h>
+#include <util/format_expr.h>
+#include <util/fresh_symbol.h>
+#include <util/pointer_expr.h>
+
+#include <analyses/local_may_alias.h>
+#include <analyses/natural_loops.h>
+#include <goto-instrument/contracts/utils.h>
+#include <langapi/language_util.h>
```

Comment:
Why is this needed?

Reply:
`from_expr` is used in this file.

Reply:
`from_expr` is for program expressions, and `format` is for logic -- we could consider renaming these to make this difference more clear.